ckanext/datastore/backend/postgres.py

```diff
@@ -700,11 +703,15 @@ def _drop_indexes(context: Context, data_dict: dict[str, Any],
             AND idx.indisprimary = false
             AND t.relname = %s
         """.format(unique='true' if unique else 'false')
-    indexes_to_drop = context['connection'].execute(
-        sql_get_index_string, data_dict['resource_id']).fetchall()
-    for index in indexes_to_drop:
-        context['connection'].execute(
-            sql_drop_index.format(index[0]).replace('%', '%%'))
+    with context['connection'].begin():
+        indexes_to_drop = context['connection'].execute(
+            sql_get_index_string, data_dict['resource_id']).fetchall()
+
+    with context['connection'].begin():
```

Comment:
``with`` now self closes (commits) the now defined transaction boundary instead of assuming that it is doing the correct thing.